numba/core/runtime/_nrt_python.c

```diff
@@ -279,7 +279,10 @@ NRT_adapt_ndarray_from_python(PyObject *obj, arystruct_t* arystruct) {
     void *data;
 
     if (!PyArray_Check(obj)) {
-        return -1;
+        obj = PyObject_CallMethod(obj, "__array__", NULL);
+        if (!obj) {
+            return -1;
+        }
```

Comment:
Again, think this needs to make sure that `obj` as returned by `__array__` is actually an array.